### PR TITLE
Fixes #23367 - add 6.4 target version

### PR DESCRIPTION
**Pull request by iNecas**
For development of next Sat release.

### CODE REVIEW

File: definitions/scenarios/upgrade_to_satellite_6_4.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    upgrade_metadata do
+      description 'Checks before upgrading to Satellite 6.4'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+      add_step(Checks::RemoteExecution::VerifySettingsFileAlreadyExists.new)
```
Comment (kgaikwad) [inline definitions/scenarios/upgrade_to_satellite_6_4.rb]: For this check, we have added [confine](https://github.com/theforeman/foreman_maintain/blob/master/definitions/checks/remote_execution/verify_settings_file_already_exists.rb#L8) block which checks for `downstream` feature plus condition `feature(:downstream).current_minor_version == '6.2'`.

If this check required here for sat 6.4 then we will need to modify this condition.

Comment (iNecas) [reply definitions/scenarios/upgrade_to_satellite_6_4.rb]: Good point, removed. Not needed after it was migrated already

**Comment by ntkathole** [comment]
@iNecas if you want to cover this BZ as well https://bugzilla.redhat.com/show_bug.cgi?id=1571368

**Comment by iNecas** [comment]
Good idea, updated

**Comment by ntkathole** [comment]
@iNecas might be we need some more checks and modifications regarding puppet. As satellite 6.3. puppet3 -> sat6.4 update failed due to dependency
```
--> Finished Dependency Resolution
Error: Package: puppet-server-3.8.6-4.el7sat.noarch (@rhel-7-server-satellite-6.3-rpms)
           Requires: puppet = 3.8.6-4.el7sat
           Removing: puppet-3.8.6-4.el7sat.noarch (@rhel-7-server-satellite-6.3-rpms)
               puppet = 3.8.6-4.el7sat
           Obsoleted By: puppet-agent-5.5.0-1.el7sat.x86_64 (sat6)
               puppet >= 4.0.0-1.el7sat
```
and upgrade path will be sat 63 with puppet4 -> sat64. So possible twicks can be if 6.3 with puppet3, raise warning (do not allow ) upgrade to 6.4 or add support to f-m to update sat 63 to puppet4.

**Comment by iNecas** [comment]
Can we handle that in separate PR?

**Comment by iNecas** [comment]
Please raise a BZ for that

**Comment by ntkathole** [comment]
@iNecas yes sure. Here it is https://bugzilla.redhat.com/show_bug.cgi?id=1571859

**Comment by iNecas** [comment]
I hear ack then? @kgaikwad mind having a final look and merging eventually?

**Comment by kgaikwad** [comment]
@iNecas, 
Ack, good to go :thumbsup:

**Comment by kgaikwad** [comment]
Thank you @iNecas & @ntkathole

**Pull request merged**